ADAL/src/ADAuthenticationContext.m

```diff
@@ -234,7 +236,17 @@ + (BOOL)isResponseFromBroker:(NSString *)sourceApplication
     isBroker = isBroker || [sourceApplication isEqualToString:ADAL_BROKER_APP_BUNDLE_ID_DOGFOOD];
 #endif
     
-    return response && isBroker;
+    NSURLComponents *components = [NSURLComponents componentsWithURL:response resolvingAgainstBaseURL:NO];
+    NSString *qp = [components percentEncodedQuery];
+    NSDictionary *queryParamsMap = [NSDictionary adURLFormDecode:qp];
+    
+    NSString *protocolVersion = queryParamsMap[@"msg_protocol_ver"];
+    BOOL isValidVersion = [protocolVersion isEqualToString:@"2"];
```

Comment:
nit: can we declare it as a constant somewhere?

Reply:
done